docs/source/core_concepts/glossary.rst

```diff
@@ -1,51 +1,81 @@
+
+Block
+=====
+
+Transaction data is permanently recorded in files called blocks.
+Blocks are organized into a linear sequence over time (also known as the block chain) [#f1]_.
+
+Blocks are signed with the cryptographic signatures of Iroha `peers <#peer>`__, voting for this block during `consensus <#consensus>`__. Signable content is called payload, so the structure of a block looks like this:
+
+*Outside payload*
+
+    - hash — SHA3-512 hash of block protobuf payload
+    - signatures — signatures of peers, which voted for the block during consensus round
+
+*Inside payload*
+
+    - height — a quantity of blocks in the chain up to the block
+    - timestamp — unix time (in millis) of block forming by a peer
+    - body — transactions, which successfully passed validation and consensus step
+    - transactions quantity
+    - previous hash of block
 
 Block Creator
 =============
 
-Components that transforms a set of transactions that have been passed `stateless <#stateless-validation>`__ and `stateful <#stateful-validation>`__ validation into a block for further propogation to `consensus <#consensus>`__.
+System component that forms a block from a set of transactions that have been passed `stateless <#stateless-validation>`__ and `stateful <#stateful-validation>`__ validation for further propagation to `consensus <#consensus>`__.
 
 Client
 ======
 
 Any application that uses Iroha is treated as a client.
 
+A distinctive feature of Iroha is that all clients are using simple client-server abstractions
+when they interact with a peer network: they don't use any abstractions which are specific for blockchain-related systems.
+For example, in Bitcoin clients have to validate blocks, or in Fabric they need to poll several peers to make sure that a transaction was written in a block,
+whereas in Iroha a client interacts with any peer similarly to a single server.
+
 Command
 =======
 
 Command is a request to Iroha that **does** change the `state <#world-state-view>`__.
-All modifications are performed via commands.
 For example, in order to create a new `role <#role>`__ in Iroha you have to issue `Create role <../api/commands.html#create-role>`__ command.
 
 Consensus
 =========
 
-Internal component that preserves consistent state among the `peers <#peer>`__ within an Iroha cluster.
-Iroha uses own consensus algorithm called Yet Another Consensus (aka YAC).
-YAC able to apply only the latest block to the chain.
-If there are missing blocks, they will be downloaded from another peers via `Synchronizer <#synchronizer>`__.
+A consensus algorithm is a process in computer science used to achieve agreement on a single data value among distributed processes or systems. Consensus algorithms are designed to achieve reliability in a network involving multiple unreliable nodes. Solving that issue -- known as the consensus problem -- is important in distributed computing and multi-agent systems.
```

Comment:
algorithm cannot be a process, it is a ordered set of steps/instruction/whatever you like